client/src/main/java/org/apache/rocketmq/client/impl/producer/DefaultMQProducerImpl.java

```diff
@@ -559,7 +559,7 @@ private SendResult sendDefaultImpl(
             MessageQueue mq = null;
             Exception exception = null;
             SendResult sendResult = null;
-            int timesTotal = communicationMode == CommunicationMode.SYNC ? 1 + this.defaultMQProducer.getRetryTimesWhenSendFailed() : 1;
```

Comment:
(1) Async Retry used another parameter for retry when send failed: retryTimesWhenSendAsyncFailed
(2) CommunicationMode has there value: SYNC, ASYNC, ONEWAY, and send the message will return directly when using the last two values. so this modification seems has no effect.

Reply:
@duhenglucky  So this pr has no effect,  maybe we should close pr?